CMakeLists.txt

```diff
@@ -16,7 +16,16 @@ set(VM_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src/vm)
 
 if(CMAKE_SYSTEM_NAME STREQUAL Linux)
     set(CLR_CMAKE_PLATFORM_UNIX 1)
-    set(CLR_CMAKE_PLATFORM_UNIX_TARGET_AMD64 1)
+    if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64)
+        set(CLR_CMAKE_PLATFORM_UNIX_TARGET_AMD64 1)
+    elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l)
```

Comment:
Just to add some real world data. Raspbian (which is hard-float) running on a classic Raspberry pi (1) gives the following output:

````bash
$ uname -a
Linux Applepi 3.18.7+ #755 PREEMPT Thu Feb 12 17:14:31 GMT 2015 armv6l GNU/Linux
````

Note there is no `armv6hl` in the output. Not sure if this is enough data to base a generalization on, but at least it's data :)